Realm/RealmSwift/SortDescriptor.swift

```diff
@@ -0,0 +1,61 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2014 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+
+/**
+A `SortDescriptor` stores a property name and a sort order for use with
+`sorted(sortDescriptors:)`. It is similar to `NSSortDescriptor`, but supports
+only the subset of functionality which can be efficiently run by Realm's query
+engine.
+*/
+public struct SortDescriptor {
```

Comment:
So far, I've only made things `Equatable` that were already equatable in Objective-C. If we make things equatable on the Swift side, then we should mirror that on ObjC side, which is out of scope for this IMO.